### PR TITLE
frontend/#4273/feat added pickUp far from current location pop up

### PR DESCRIPTION
**Pull request by MercyQueen**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

https://github.com/nammayatri/nammayatri/assets/55204010/2674ff64-d71c-48a0-ab0f-d4f626e50298

### CODE REVIEW

File: Frontend/ui-customer/src/Flow.purs [hunk 1]
```diff
@@ -1535,19 +1541,25 @@ rideSearchFlow flowType = do
...
                   if ((MU.getMerchant FunctionCall) /= MU.YATRI && response.distance >= 50000 )then do
                     _ <- pure $ updateLocalStage DistanceOutsideLimits
                     modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen{props{currentStage = DistanceOutsideLimits ,rideRequestFlow = true, isSearchLocation = SearchLocation}})
                     void $ lift $ lift $ toggleLoader false
-                    else if response.distance < 500 && (getValueToLocalStore LOCAL_STAGE) /= "ShortDistance" then do
-                      _ <- pure $ updateLocalStage ShortDistance
-                      modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen{props{currentStage = ShortDistance ,rideRequestFlow = true, isSearchLocation = SearchLocation, distance = response.distance}})
+                    else if response.distance < 500 && (getValueToLocalStore LOCAL_STAGE) /= "ShortDistance" && getValueToLocalStore LOCAL_STAGE /= "PickUpFarFromCurrentLocation" then do
```
Comment (ashishsingh101) [inline Frontend/ui-customer/src/Flow.purs:1549]: if you want we can simplify this approach using inner hierarchical conditions or cases

**Review by ashishsingh101 — CHANGES_REQUESTED** [review]

File: Frontend/ui-customer/src/Flow.purs [hunk 2]
```diff
@@ -749,15 +749,21 @@ homeScreenFlow = do
...
           if ((MU.getMerchant FunctionCall) /= MU.YATRI && response.distance >= 50000) then do
             updateLocalStage DistanceOutsideLimits
-            modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen{props{currentStage = DistanceOutsideLimits ,rideRequestFlow = true, isSearchLocation = SearchLocation, findingQuotesProgress = 0.0}})
+            modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen{props{currentStage = DistanceOutsideLimits ,rideRequestFlow = true, isSearchLocation = SearchLocation, findingQuotesProgress = 0.0, shorterTrip = false}})
             homeScreenFlow
-            else if response.distance < 500 && getValueToLocalStore LOCAL_STAGE /= "ShortDistance" then do
-              updateLocalStage ShortDistance
-              modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen{props{currentStage = ShortDistance ,rideRequestFlow = true, isSearchLocation = SearchLocation, distance = response.distance, findingQuotesProgress = 0.0}})
+            else if response.distance < 500 && getValueToLocalStore LOCAL_STAGE /= "ShortDistance" && getValueToLocalStore LOCAL_STAGE /= "PickUpFarFromCurrentLocation" then do
```
[the remark this answers is not in the document]
Comment (MercyQueen) [reply Frontend/ui-customer/src/Flow.purs:757]: It can be replaced with `all`.

File: Frontend/ui-customer/src/Flow.purs [hunk 3]
```diff
@@ -749,15 +749,18 @@ homeScreenFlow = do
...
             updateLocalStage DistanceOutsideLimits
-            modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen{props{currentStage = DistanceOutsideLimits ,rideRequestFlow = true, isSearchLocation = SearchLocation, findingQuotesProgress = 0.0}})
+            modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen{props{currentStage = DistanceOutsideLimits ,rideRequestFlow = true, isSearchLocation = SearchLocation, findingQuotesProgress = 0.0, shorterTrip = false}})
             homeScreenFlow
-            else if response.distance < 500 && getValueToLocalStore LOCAL_STAGE /= "ShortDistance" then do
-              updateLocalStage ShortDistance
-              modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen{props{currentStage = ShortDistance ,rideRequestFlow = true, isSearchLocation = SearchLocation, distance = response.distance, findingQuotesProgress = 0.0}})
+            else if ( (response.distance < 500  || isDistMoreThanThreshold )&& Arr.all (_ /= getValueToLocalStore LOCAL_STAGE ) ["PickUpFarFromCurrentLocation", "ShortDistance"]) then do
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Flow.purs:758]: try to use isLocalStageOn function to check currentStage. Instead of comparing hard coded string.

**Review by Vignesh-772 — CHANGES_REQUESTED** [review]
Resolve comments

**Review by Vignesh-772 — DISMISSED** [review]

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs [hunk 4]
```diff
@@ -152,6 +152,9 @@ screen initialState =
               SettingPrice -> do
                 _ <- pure $ removeMarker (getCurrentLocationMarker (getValueToLocalStore VERSION_NAME))
                 pure unit
+              PickUpFarFromCurrentLocation -> do 
+                _ <- pure $ removeMarker (getCurrentLocationMarker (getValueToLocalStore VERSION_NAME))
```
Comment (witcher-shailesh) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs:156]: replace this also with `void $` and try to correct the same in other functions

**Review by ashishsingh101 — DISMISSED** [review]

**Review by witcher-shailesh — APPROVED** [review]

**Review by ashishsingh101 — APPROVED** [review]

**Pull request merged**